### PR TITLE
Refactor tools/tidy

### PR DESCRIPTION
**Pull request by perlpunk**
So that it can be made equal to os-autoinst/tools/tidy

Issue: https://progress.opensuse.org/issues/70384

PR for os-autoinst: https://github.com/os-autoinst/os-autoinst/pull/1518

### CODE REVIEW

File: tools/tidy [hunk 1]
```diff
@@ -32,62 +39,56 @@ while true; do
...
-        if ! [ $only_changed ] || [ "$(git diff --name-only "$file")" ]; then
-            changed_files+=("$file")
-        fi
-    done
+find-files() {
+    local files=('**.p[ml]' '**.t')
+    if [[ -d script ]]; then
+        files+=($(file --mime-type script/*| grep perl | awk -F':' '{ print $1 }'))
+    fi
+    if [[ -f isotovideo ]]; then
```
Comment (kalikiana) [inline tools/tidy:63]: Maybe we can go by `file --mime-type` here, too?

**Review by kalikiana — COMMENTED** [review]
I like it, specifically because it's a speed boost as well by replacing `find` by git ls-files`

File: tools/tidy (hunk 1 above)
Comment (perlpunk) [reply tools/tidy:63]: What do you mean? Something like
`file --mime-type *`
?

Comment (kalikiana) [reply tools/tidy:63]: We can assume if it's `text-x-perl` we want to check it, like the `script` folder just above, no?

**Comment by perlpunk** [comment]
> I like it, specifically because it's a speed boost as well by replacing `find` by git ls-files`

Actually it is significantly slower than before, and I'm not sure why.
in openQA master it takes 50s, with my PR 1:30.

For os-autoinst: master 10s, PR 28s

File: tools/tidy (hunk 1 above)
Comment (perlpunk) [reply tools/tidy]: Ok, I rewrote that part.

**Comment by perlpunk** [comment]
I think I found out why it is slower.
```
% git ls-files '**.p[ml]' '**.t' isotovideo  | time xargs perltidy --pro=.../.perltidyrc
xargs perltidy --pro=.../.perltidyrc  9.12s user 0.02s system 99% cpu 9.141 total
% git ls-files '**.p[ml]' '**.t' isotovideo  | time xargs -I {} perltidy --pro=.../.perltidyrc "{}"
xargs -I {} perltidy --pro=.../.perltidyrc "{}"  25.71s user 1.45s system 99% cpu 27.189 total
```

Does anyone know if we need the `-I {}`?

**Review by kalikiana — APPROVED** [review]
I'm liking it. Thanks for bearing with my confusing questions 😉

**Review by okurz — APPROVED** [review]
I verified locally that the runtime is fine.

**Pull request merged**